bazel/cc_toolchains/clang_configuration.bzl

```
@@ -26,18 +26,23 @@ def _detect_system_clang(repository_ctx):
     # If the user provides an explicit `CC` environment variable, use that as
     # the compiler. This should point at the `clang` executable to use.
     cc = repository_ctx.os.environ.get("CC")
+    cc_path = None
     if cc:
-        version_output = _run(repository_ctx, [cc, "--version"]).stdout
-        if not "clang" in version_output:
-            fail("The `CC` environment variable is not a Clang compiler.")
-        return repository_ctx.path(cc)
-
-    # Try looking on the path. We only check for the normal name.
-    system_clang = repository_ctx.which("clang")
-    if not system_clang:
+        cc_path = repository_ctx.path(cc)
+        if not cc_path.exists:
+            cc_path = repository_ctx.which(cc)
```

```suggestion
        cc_path = repository_ctx.which(cc)
```

Please double-check this, but I've been thinking about this and I don't think the double-check is needed. And actually, relying on `.which` should implicitly check `.exists`, so we can probably simplify to just `if not cc_path` without checking `exists` as well.

---

I double checked this and all three of these have different effects:
```
if not cc_path:
    cc_path = repository_ctx.which(cc)
```
```
if not cc_path.exists:
    cc_path = repository_ctx.which(cc)
```
```
cc_path = repository_ctx.which(cc)